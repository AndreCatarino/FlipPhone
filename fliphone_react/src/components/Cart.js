import React, {Component} from 'react';
import axios from "axios";
import {Button, Card, Col, Row} from "react-bootstrap";



class Cart extends Component {

    constructor(props) {
        super(props);
        this.state = {
            items: [],
            // params necessary for payment post
            order_id:'',
            dformat:'',
            userID:'',
            seller_ids:'',
            checkout:''
        }
    }

    allStorage() {
        //console.log("function trigered");
        let userID = localStorage.getItem("userID");
        console.log(userID);
        let values = [],
            keys = Object.keys(localStorage),
            i = keys.length;

        while ( i-- ) { // implement an if statement to check for the user_id item, so we dont include it in values list

            if (localStorage.getItem(keys[i]) != userID){
                values.push( localStorage.getItem(keys[i]) );
            }

        }
        console.log(values);
        return values;

    }


    componentDidMount() {
        let itemID_list = this.allStorage();
        //console.log(itemID_list);
        let itemID_Set = new Set(itemID_list); // for unique ID's
        //console.log(itemID_Set);
        let newState = Object.assign({}, this.state); // Clone the state obj in newState
        //console.log(newState);
        for(let id of itemID_Set) {
            console.log(id);
            axios.get('http://localhost:8080/item/' + id)
                .then(response => response.data)
                .then((data) => {
                    newState['items'].push(data); // push to newState the requested json data
                    this.setState(newState);
                });

        }


    }



    removeItem = (param) => {
        console.log(param);
        alert("Item removido do carrinho");


        localStorage.removeItem("item"+param); // remove from local cache (if the page reloads the items in the cart will dissapear)


        //next task its to remove item from this.state.items so it isnt rendered anymore

        //let cloneState = Object.assign({}, this.state);
        //console.log(cloneState);
        for(let i in this.state.items) {
            //console.log(i);
            if (this.state.items[i].item_id === param){
                //console.log("hit");
                let newState = this.state.items.pop(i);
                //console.log(newState);
                this.setState(newState) ;
            }
        }

    }



    BuyItems = (items) => { // post an order & post a payment

        console.log(items);
        let TotalPrice = 0;
        let items_ids = [];
        let seller_ids = []; // to post on payment endpoint
        for (let i in items){
            let price = parseFloat(items[i].price);
            TotalPrice += price;
            items_ids.push(items[i].item_id);
            seller_ids.push(items[i].seller_id.user_id);
        }

        console.log(TotalPrice);
        console.log(items_ids);
        //console.log(order_ids);
        console.log(seller_ids);

        let d = new Date,
            dformat = [d.getMonth()+1,
                    d.getDate(),
                    d.getFullYear()].join('/')+' '+
                [d.getHours(),
                    d.getMinutes(),
                    d.getSeconds()].join(':');


        console.log(dformat);
        let userID = localStorage.getItem("userID");



        axios.post("http://localhost:8080/order/add?date="+dformat+"&total="+TotalPrice+"&client_id="+userID+"&item_id="+items_ids)
            .then(response => {
                if(response.data == null){
                   alert("An error has occurred during the transaction");
                }

            })

        //get order id
        let highest_order_id=0;
        axios.get('http://localhost:8080/order/all')// + id)
            .then(response => response.data)
            .then((data) => {
                console.log(data);
                for (let i in data){
                    //console.log(data[i]);
                    if (data[i].order_id > highest_order_id){
                        highest_order_id=data[i].order_id;
                    }
                }
                //console.log(highest_order_id);
                this.setState({'order_id':highest_order_id});
            });


        // set the right params for the payment post
        this.setState({'seller_ids':seller_ids});
        this.setState({'dformat':dformat});
        this.setState({'userID':userID});
        this.setState({'checkout':1});




        //clear all items
        localStorage.clear();

        //redirect to main page
        // the user will have to login again
        window.location.replace("http://localhost:3000/");
    }


    PostPayment= () => {

        console.log(this.state.order_id,this.state.userID,this.state.seller_ids, this.state.dformat, this.state.seller_ids);

        let order_id= parseInt(this.state.order_id);
        let client_id=parseInt(this.state.userID);

        axios.post("http://localhost:8080/payment/add?status="+"SENT"+"&gateway="+"PAYPAL"+'&date='+this.state.dformat+"&order_id="+ order_id +"&client_id="+client_id+"&seller_id="+this.state.seller_ids)
            .then(response => {
                if(response.data != null){
                    alert("Buy order successfull");
                    console.log(this.state.order_id,this.state.userID,this.state.seller_ids, this.state.dformat,this.state.order_id, this.state.seller_ids);
                }
                else{
                    alert("An error has occurred during the transaction");
                }

            })
    }



    render() {

        let items = this.state.items;
        console.log(items);
        console.log(this.state.order_id);

        console.log(this.state.checkout);

        if (items.length === 0) {
            return (<p>The cart is empty</p>);
        }



        if (this.state.checkout==1){

            return(
                <div>
                    <select  >
                        <option></option>
                    </select>
                    <Button onClick={() => this.PostPayment()}  primary enabled>
                        Confirm purchase
                    </Button>
                </div>
            );
        }



        let RenderItems = this.state.items.map(item => {
            let id = item.item_id;

            return (
                    <Row className='show-grid'>
                        <Col xs={4} md={2}> color : {item.color}</Col>
                        <Col xs={4} md={2}> price : {item.price}</Col>
                        <Col xs={4} md={2}> grade : {item.grade}</Col>
                        <Col xs={4} md={2}> version : {item.version}</Col>
                        <Col xs={5} md={4}>
                            <Button  variant={"info"} onClick={() => this.removeItem(id)}>
                                Remove from Cart
                            </Button>
                        </Col>

                    </Row>
            )
        })

        return (

            <Card style={{borderTop: '1px solid #f2f2f2'}}>
                {RenderItems}
                <Button onClick={() => this.BuyItems(this.state.items)}  primary enabled>
                    Buy Items
                </Button>
            </Card>

        )


    }
}

export default Cart;
