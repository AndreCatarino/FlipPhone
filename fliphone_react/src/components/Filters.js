import React, {Component} from 'react';
import {Card, Table} from 'react-bootstrap';

import {faList} from '@fortawesome/free-solid-svg-icons';
import axios from 'axios';



export default class Filters extends Component {

    constructor(props){
        super(props);
        this.state = {
            phones : [],
            searchByName : '',
            searchByCPU_GPU : '',
            searchByBattery : '',
            searchByImage : '',

        };
    }

    componentDidMount() {
        axios.get("http://localhost:8080/product/all")
            .then(response => response.data)
            .then((data) => {
            this.setState({phones: data});
            //console.log(this.state.phones);
            });

    }



    updateSearchByName(event) {
        this.setState({searchByName: event.target.value});
    }

    updateSearchByCPU_GPU(event) {
        this.setState({searchByCPU_GPU: event.target.value});
    }

    updateSearchByBattery(event) {
        this.setState({searchByBattery: event.target.value});
    }

    updateSearchByImage(event) {
        this.setState({searchByImage: event.target.value});
    }


    render() {

        // not case sensitive
        let filteredProducts = this.state.phones.filter(
            (phone) => { // return the result of more than 1 filter --> by name, by cpu_gpu, by battery, and by image
                return phone.product_name.toLowerCase().indexOf(this.state.searchByName.toLowerCase()) !== -1 &&
                    phone.cpu_gpu.toLowerCase().indexOf(this.state.searchByCPU_GPU.toLowerCase()) !== -1 &&
                    phone.battery.toLowerCase().indexOf(this.state.searchByBattery.toLowerCase()) !== -1 &&
                    phone.image.toLowerCase().indexOf(this.state.searchByImage.toLowerCase()) !== -1;

            }
        );
        const elementStyle ={
            border:'solid',
            borderRadius:'10px',
            position:'relative',
            left:'0vh',
            height:'3vh',
            width:'20vh',
            marginTop:'0vh',
            marginBottom:'0vh'
        }

        return(

            <div>
                <p>Product filters (case insensitive):</p>
                <input type="text" placeholder="Search by product name"  style={elementStyle} value={this.state.searchByName} onChange={this.updateSearchByName.bind(this)}/>
                <input type="text" placeholder="Search by cpu_gpu"  style={elementStyle} value={this.state.searchByCPU_GPU} onChange={this.updateSearchByCPU_GPU.bind(this)}/>
                <input type="text" placeholder="Search by battery"  style={elementStyle} value={this.state.searchByBattery} onChange={this.updateSearchByBattery.bind(this)}/>
                <input type="text" placeholder="Search by image"  style={elementStyle} value={this.state.searchByImage} onChange={this.updateSearchByImage.bind(this)}/>

                <Card className={"border border-dark bg-dark text-white"}>
                <Card.Header> All products: ({this.state.phones.length} products available in total) </Card.Header>

                <Card.Body>
                    <Table striped bordered hover variant="dark" >
                        <thead>
                        <tr>
                            <th>product_name</th>
                            <th>cpu_gpu</th>
                            <th>ram_rom</th>
                            <th>image</th>
                            <th>screen_size</th>
                            <th>screen_type</th>
                            <th>battery</th>
                            <th>os</th>
                            <th>selfie_cam</th>
                            <th>camera</th>
                        </tr>
                        </thead>
                        <tbody>
                        {filteredProducts.length === 0 ?
                            <tr align={'center'}>
                                <td colSpan={"10"}> No Products Available</td>
                            </tr> :
                            filteredProducts.map((phone) => (
                               <tr key={phone.id}>
                                   <td>{phone.product_name}</td>
                                   <td>{phone.cpu_gpu}</td>
                                   <td>{phone.ram_rom}</td>
                                   <td>{phone.image}</td>
                                   <td>{phone.screen_size}</td>
                                   <td>{phone.screen_type}</td>
                                   <td>{phone.battery}</td>
                                   <td>{phone.os}</td>
                                   <td>{phone.selfie_cam}</td>
                                   <td>{phone.camera}</td>
                               </tr>
                            ))


                        }
                        </tbody>
                    </Table>
                </Card.Body>
            </Card>
            </div>
        )


    }
}


