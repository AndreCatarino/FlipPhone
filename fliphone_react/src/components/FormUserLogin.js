import React, { Component } from 'react';
import Dialog from '@material-ui/core/Dialog';
import AppBar from '@material-ui/core/AppBar';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import axios from "axios";
import {Form} from "react-bootstrap";

export class FormUserLogin extends Component {
    continue = e => {
        e.preventDefault();
        //this.props.nextStep();
    };

    constructor(props){
        super(props);
        this.state = { password:'',name:'', salt:'',email:'',address:'',nif:'',type:''}
        this.LoginDataChange = this.LoginDataChange.bind(this);
        this.submitLoginData = this.submitLoginData.bind(this);
    }




    initialState = {
        password:'',name:'', salt:'', email:'',address:'',nif:'',type:''
    }

    resetLoginData = () => {
        this.setState(() => this.initialState)
    }


    submitLoginData = event => {
        //alert("password: "+this.state.password+" name: "+this.state.name+" salt: "+this.state.salt+ "email:" + this.state.email + " address: " + this.state.address + " nif: "+this.state.nif + " type: " + this.state.type  );
        event.preventDefault();
        //alert("Register successfull");
        // need to store locally the user id (used in order ID)
        const login = {

            password : this.state.password,
            name : this.state.name,
            salt : this.state.salt,
            email : this.state.email,
            address : this.state.address,
            nif : this.state.nif,
            type : this.state.type

        };

        axios.post("http://localhost:8080/user/add?password="+this.state.password+"&name="+this.state.name+"&salt="+"salt"+"&email="+this.state.email+"&address="+this.state.address+"&nif="+this.state.nif+"&type="+this.state.type)
            .then(response => {
                if(response.data != null){
                    this.setState(this.initialState);
                    alert("Register successfull");
                    window.location.replace("http://localhost:3000/main"); //cant go back from this url
                }
            })

    }


    LoginDataChange= event => {
        this.setState({
            [event.target.name]:event.target.value
        });
    }

    render() {
        //const { values, handleChange } = this.props;

        const {password,name, salt, email,address,nif,type} = this.state;


        return (
            <MuiThemeProvider>
                <>
                    <Dialog
                        open
                        fullWidth
                        maxWidth='sm'
                    >

                        <Form  onReset={this.resetLoginData} onSubmit={this.submitLoginData} >
                        <TextField
                            required
                            autoComplete="off"
                            placeholder="Enter Your name"
                            label="name"
                            type="text"
                            name="name"
                            value={name}
                            onChange={this.LoginDataChange}
                            margin="normal"
                            fullWidth
                        />
                        <br />

                        <TextField
                            required
                            autoComplete="off"
                            placeholder="Enter Your address"
                            label="address"
                            type="text"
                            name="address"
                            value={address}
                            onChange={this.LoginDataChange}
                            margin="normal"
                            fullWidth
                        />
                        <br />

                        <TextField
                            required
                            autoComplete="off"
                            placeholder="Enter Your nif"
                            label="nif"
                            type="text"
                            name="nif"
                            value={nif}
                            onChange={this.LoginDataChange}
                            margin="normal"
                            fullWidth
                        />
                        <br />

                        <TextField
                            required
                            autoComplete="off"
                            placeholder="Enter Your type"
                            label="type"
                            type="text"
                            name="type"
                            value={type}
                            onChange={this.LoginDataChange}
                            margin="normal"
                            fullWidth
                        />
                        <br />

                        <TextField
                            required
                            autoComplete="off"
                            placeholder="Enter Your Email"
                            label="Email"
                            type="text"
                            name="email"
                            value={email}
                            onChange={this.LoginDataChange}
                            margin="normal"
                            fullWidth
                        />
                        <br />
                        <TextField
                            required
                            autoComplete="off"
                            placeholder="Enter Your Password"
                            label="Password"
                            type="text"
                            name="password"
                            value={password}
                            onChange={this.LoginDataChange}
                            margin="normal"
                            fullWidth
                        />
                        <br />
                        <div style={{"textAlign":"center"}}>
                            <Button
                                    size="sm"
                                    color="primary"
                                    variant="contained"
                                    type="submit"

                            >Register</Button>
                            <Button
                                size="sm"
                                color="primary"
                                variant="info"
                                type="reset"

                            >Reset</Button>
                        </div>

                    </Form>

                    </Dialog>
                </>
            </MuiThemeProvider>
        );
    }
}
