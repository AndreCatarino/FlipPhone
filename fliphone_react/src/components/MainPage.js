import React, {Component} from 'react';
import {CardColumns,Card,Row,Col,Container} from 'react-bootstrap';
import axios from "axios";

import PhoneCard from "./PhoneCards";

class MainPage extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            phones: []
        }
        global.MyVar = 'test';
    }


    componentDidMount() {
        axios.get("http://localhost:8080/product/all")
            .then(response => response.data)
            .then((data) => {
                this.setState({phones: data});
                //console.log(this.state.phones);
            });


    }



    render() {
        let Products = this.state.phones;
        //console.log(Products);




        let phoneCards = this.state.phones.map(phone => {
            return (
                <Col sm={4}>
                    <PhoneCard phone={phone} />
                </Col>

                /*
                col {
                    -ms-flex-preferred-size: 0;
                    flex-basis: 0;
                    -ms-flex-positive: 1;
                    flex-grow: 1;
                    max-width: 30%;
                    padding-right: 20px;
                }
                */
            )
        })



        const linkStyle ={
            color:'darkgray',
            fontSize:'23px'

        }

        return (
            <div>
                <a style={linkStyle} href="/list">Search for your desired phone with our filters here</a>
                <p></p>
            <Container fluid>
                <Row>
                    {phoneCards}
                </Row>

            </Container>
            </div>
        )
    }
}

export default MainPage;
