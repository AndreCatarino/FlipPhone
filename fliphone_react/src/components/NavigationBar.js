import React from 'react';
import {Navbar, Nav} from 'react-bootstrap';
import {Link} from 'react-router-dom'


class NavigationBar extends React.Component {
    render () {
        return (
            <Navbar bg="dark" variant="dark">
                <Link to={"/main"} className="navbar-brand">
                    <img src="https://cdn.pocket-lint.com/r/s/320x/assets/images/138137-phones-buyer-s-guide-which-is-the-best-mid-range-phone-under-400-image1-w3pdengkrc.jpg?v1" width="100" height="60" alt="brand"/>
                    FlipPhone WebApp
                </Link>

                <Nav className="mr-auto">
                    <Link to={"/add"} className="nav-link">Sell Phones</Link>
                    <Link to={"/list"} className="nav-link">Filters</Link>
                    <Link to={"/users"} className="nav-link">User List</Link>
                    <Link to={"/cart"} className="nav-link">Cart</Link>
                    <Link to={"/logout"} className="nav-link">Logout</Link>
                </Nav>
                </Navbar>
        );
    }
    
}

export default NavigationBar;

