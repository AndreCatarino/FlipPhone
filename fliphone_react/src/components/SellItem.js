import React, {Component} from 'react'
import {Card, Form, Button, Col, Row} from 'react-bootstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faList} from '@fortawesome/free-solid-svg-icons';
import axios from 'axios';

export default class Phone extends Component{

    constructor(props){
        super(props);
        this.state = {
            products:[],
            grade:'', color:'', price:'', version:'',product_id:'',seller_id:''
        }
        this.phoneChange = this.phoneChange.bind(this);
        this.submitPhone = this.submitPhone.bind(this);
    }



    componentDidMount() {
        axios.get("http://localhost:8080/product/all")
            .then(response => response.data)
            .then((data) => {
                this.setState({products: data});
                //console.log(data[0].product_id);
                this.setState({product_id:data[0].product_id}) //defining default value for product id (id of the first product) because if the user doesnt choose any option in the select,the value for product_id would be undefined
            });

        //console.log(this.state.products);
        let userID = localStorage.getItem("userID");
        this.setState({seller_id:userID});
    }



    initialState = {
        grade:'', color:'', price:'', version:'',product_id:'',seller_id:''
    }




    resetSellOrder = () => {
        this.setState(() => this.initialState)
    }



    submitPhone = event => {
        //alert("grade: "+ this.state.grade +" color: "+this.state.color+" price: "+this.state.price+ "version:" + this.state.version + "product_id:" + this.state.product_id + " seller_id:"+this.state.seller_id );
        event.preventDefault();

        let price = parseFloat(this.state.price);
        let product_id = parseInt(this.state.product_id);
        console.log(product_id);
        let seller_id = parseInt(this.state.seller_id);
        console.log(seller_id);

        axios.post("http://localhost:8080/item/add?grade="+this.state.grade+"&color="+this.state.color+"&price="+price+"&version="+this.state.version+"&product_id="+product_id+"&seller_id="+seller_id)
            .then(response => {
                if(response.data != null){
                    this.setState(this.initialState);
                    alert("Sell order successfull");
                    window.location.replace("http://localhost:3000/main"); // redirect to main page
                }

            })

    }



    phoneChange= event => {
        this.setState({
            [event.target.name]:event.target.value
        });
    }


    selectChange= event => {
        //alert(event.target.value);
        this.setState({'product_id':event.target.value})
    }



    render() {
        //console.log(this.state.products);
        const {grade, color, price, version,product_id,seller_id} = this.state;
        // get client id for seller_id parameter

        //let counter = -1
        let RenderProducts = this.state.products.map(product => {
            //counter += 1;
            return (
                <option value={product.product_id}>{product.product_name}</option>
            )})


        return (
            <Card className={"border border-dark bg-dark text-white"}>
                <Card.Header> <FontAwesomeIcon icon={faList } /> Selling Details </Card.Header>
                <Form onReset={this.resetSellOrder} onSubmit={this.submitPhone} id="phoneFormId">
                    <Card.Body>
                        <Form.Row>
                            <Form.Group as={Col} controlId="formGridName">
                                <Form.Label> Phone's grade </Form.Label>
                                <Form.Control required autoComplete="off" type="text" name="grade" value={grade} onChange={this.phoneChange} className={"bg-dark text-white"} placeholder="Enter your phone's grade" />
                            </Form.Group>
                            <Form.Group as={Col} FontAwesomeIcon controlId="formGridPrice">
                                <Form.Label> color </Form.Label>
                                <Form.Control required autoComplete="off" type="text" name="color" value={color} onChange={this.phoneChange} className={"bg-dark text-white"} placeholder="Enter color" />
                            </Form.Group >
                        </Form.Row>
                        <Form.Row>
                            <Form.Group as={Col} controlId="formGridRam_rom">
                                <Form.Label> Phone's price </Form.Label>
                                <Form.Control required autoComplete="off" type="text" name="price" value={price} onChange={this.phoneChange} className={"bg-dark text-white"} placeholder="Enter your phone's price" />
                            </Form.Group>
                            <Form.Group as={Col} controlId="formGridImageL">
                                <Form.Label> Phone's version </Form.Label>
                                <Form.Control required autoComplete="off" type="text" name="version" value={version} onChange={this.phoneChange} className={"bg-dark text-white"} placeholder="Enter version " />
                            </Form.Group>
                        </Form.Row>
                        <Form.Row>
                            <Form.Group as={Col} controlId="formGridDescription">
                                <Form.Label> Choose in which product your phone fits </Form.Label>
                                <br/>
                                <label >Choose ProductName:</label>

                                <select onChange={this.selectChange} >
                                    {RenderProducts}
                                </select>
                            </Form.Group>

                        </Form.Row>

                    </Card.Body>
                    <Card.Footer style={{"textAlign":"center"}}>
                        <Button size="sm" variant="success" type="submit">
                            Sell Phone
                        </Button> {' '}
                        <Button size="sm" variant="info" type="reset">
                            Reset
                        </Button>
                    </Card.Footer>
                </Form>
            </Card>
        );
    }
}
