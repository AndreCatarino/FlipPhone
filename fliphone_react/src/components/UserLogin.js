import React, { Component } from 'react';
import Dialog from '@material-ui/core/Dialog';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import axios from "axios";
import {Form} from "react-bootstrap";

export class UserLogin extends Component {
    continue = e => {
        e.preventDefault();
        //this.props.nextStep();
    };

    constructor(props) {
        super(props);
        this.state = { password:'',email:''}
        this.LoginDataChange = this.LoginDataChange.bind(this);
        this.submitLoginData = this.submitLoginData.bind(this);
    }

// use axios to confirm a registered user
    // guardar na cache o user_id



    initialState = {
        password:'', email:''
    }

    resetLoginData = () => {
        this.setState(() => this.initialState)
    }

    submitLoginData = event => {
        //alert("password: "+this.state.password+" email: "+this.state.email);
        event.preventDefault();


        axios.get("http://localhost:8080/user/byEmail?email="+this.state.email) // get by email
            .then(response => response.data)
            .then((data) => {
                //this.setState({loginData: data});
                console.log(data);
                //console.log(data.user_id);
                let user_id = data.user_id;
                localStorage.setItem("userID",user_id)    //guardar na cache o user_id
                if(data.password === this.state.password){
                    alert("Login successfull");

                    //final step --> redirecionar para a main
                    window.location.replace("http://localhost:3000/main");
                }
                else if (data == []){
                    alert("email is not registered in our platform");
                }
                else{
                    alert("wrong password");
                }


            });


        //final step --> redirecionar para a main
        //window.location.replace("http://localhost:3000/main");
    }



    LoginDataChange= event => {
        this.setState({
            [event.target.name]:event.target.value
        });
    }




    render() {


        const {password, email} = this.state;


        return (
            <MuiThemeProvider>
                <>
                    <Dialog
                        open
                        fullWidth
                        maxWidth='sm'
                    >

                        <Form  onReset={this.resetLoginData} onSubmit={this.submitLoginData} >

                            <TextField
                                required
                                autoComplete="off"
                                placeholder="Enter Your Email"
                                label="Email"
                                type="text"
                                name="email"
                                value={email}
                                onChange={this.LoginDataChange}
                                margin="normal"
                                fullWidth
                            />
                            <br />

                            <TextField
                                required
                                autoComplete="off"
                                placeholder="Enter Your Password"
                                label="Password"
                                type="text"
                                name="password"
                                value={password}
                                onChange={this.LoginDataChange}
                                margin="normal"
                                fullWidth
                            />
                            <br />
                            <div style={{"textAlign":"center"}}>
                                <Button
                                    size="sm"
                                    color="primary"
                                    variant="contained"
                                    type="submit"

                                >Login</Button>
                                <Button
                                    size="sm"
                                    color="primary"
                                    variant="info"
                                    type="reset"

                                >Reset</Button>
                            </div>

                        </Form>

                    </Dialog>
                </>
            </MuiThemeProvider>
        );
    }
}
export default UserLogin;
