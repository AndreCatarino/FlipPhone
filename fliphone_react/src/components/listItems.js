import React, {Component} from 'react';
import {CardColumns, Card, Row, Col, Container, Button, Form} from 'react-bootstrap';
import axios from "axios";
//import {Button,Spinner} from 'belle';

import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import ButtonBase from '@material-ui/core/ButtonBase';
import {counter} from "@fortawesome/fontawesome-svg-core";
import PhoneCard from "./PhoneCards";

class listItems extends Component {

    constructor(props) {
        super(props);
        this.state = {
            items: [],
            CacheItems : []
        }

    }



    componentDidMount() {
        const product_id = this.props.match.params.product_id;
        //console.log(this.props.match.params.product_id);
        axios.get('http://localhost:8080/item/byProduct?product_id='+product_id)
            .then(response => response.data)
            .then((data) => {
                this.setState({items: data});
            });
    }




    addToCart = (param) => {
        //console.log(param);
        alert("Item adicionado ao carrinho");
        this.state.CacheItems.push(param);
        //let CacheItemsSet = new Set(this.state.CacheItems);

        console.log(this.state.CacheItems);

        localStorage.setItem("item"+this.state.CacheItems,this.state.CacheItems);
    }


    render() {

        var result = [];
        for( let i in this.state.items)
            result.push([i, this.state.items [i]]);

        console.log(this.state.items);


        let items = this.state.items;

        if(items.length === 0){
            return (<p>This product has no items(out of stock)</p>);
        }



        let RenderItems = this.state.items.map(item => {
            let id = item.item_id;
            //alert(id);
            return (
                    <Row className='show-grid' >
                        <Col xs={4} md={2}> color : {item.color}</Col>
                        <Col xs={4} md={2} > price :  {item.price}</Col>
                        <Col xs={4} md={2} > grade :  {item.grade}</Col>
                        <Col xs={4} md={2} > version :  {item.version}</Col>
                        <Col xs={5} md={4} >
                            <Button type={'submit'} onClick={() => this.addToCart(id)} primary enabled >
                                Add to Cart
                            </Button>
                        </Col>

                    </Row>
            )})

        return (

                <Card style={{ borderTop: '1px solid #f2f2f2' }}>
                    {RenderItems}
                </Card>

        )

    }
}



export default listItems;
