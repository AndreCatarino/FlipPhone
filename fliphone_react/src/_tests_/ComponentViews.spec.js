import { shallow, render, mount } from 'enzyme';
import React from 'react';
import SellItem from "../components/SellItem";
import Filters from "../components/Filters";
import Cart from "../components/Cart";
import UserLogin from "../components/UserLogin";
import renderer from 'react-test-renderer';
import {FormUserRegister} from "../components/FormUserRegister";



//testing the view of a component //--> next checkar value dos inputs.. quando sao enumerados
describe('SellItem', () => {
    it('renders a form', () => {
        const SellItems = shallow(<SellItem />);
        expect(SellItems.find('Form').length).toEqual(1);
    });
    it('renders a select', () => {
        const SellItems = shallow(<SellItem />);
        expect(SellItems.find('select').length).toEqual(1);
    });
    it('renders 2 buttons', () => {
        const SellItems = shallow(<SellItem />);
        expect(SellItems.find('Button').length).toEqual(2);
    });

});

describe('filters', () => {

    it('renders inputs', () => {
        const filter = shallow(<Filters />);
        expect(filter.find("input").length).toEqual(4);
    });

    it('renders Card', () => {
        const filter = shallow(<Filters />);
        expect(filter.find('Card').length).toEqual(1);
    });

});



describe('UserLogin', () => {

    it('renders  Form', () => {
        const login = shallow(<UserLogin />);
        expect(login.find('Form').length).toEqual(1);
    });


});



describe('SellItem Form Snapshot', () => { // Instead of rendering the graphical UI, which would require building the entire app, you can use a test renderer
    it('renders a form', () => {
        const SellItems = shallow(<SellItem/>);
        const tree = renderer
            .create(SellItems.find("Form").html())
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});


/*
describe('Register Snapshot', () => { // Instead of rendering the graphical UI, which would require building the entire app, you can use a test renderer
    it('renders a form', () => {
        const Register = shallow(<FormUserRegister />);
        const tree = renderer
            .create(Register.find("Form").html())
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
*/
