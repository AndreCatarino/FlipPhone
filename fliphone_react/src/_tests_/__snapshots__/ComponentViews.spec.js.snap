// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellItem Form Snapshot renders a form 1`] = `"<form id=\\"phoneFormId\\" class=\\"\\"><div class=\\"card-body\\"><div class=\\"form-row\\"><div class=\\"form-group col\\"><label class=\\"form-label\\" for=\\"formGridName\\"> Phone&#x27;s grade </label><input type=\\"text\\" required=\\"\\" autoComplete=\\"off\\" name=\\"grade\\" value=\\"\\" placeholder=\\"Enter your phone&#x27;s grade\\" id=\\"formGridName\\" class=\\"bg-dark text-white form-control\\"/></div><div class=\\"form-group col\\"><label class=\\"form-label\\" for=\\"formGridPrice\\"> color </label><input type=\\"text\\" required=\\"\\" autoComplete=\\"off\\" name=\\"color\\" value=\\"\\" placeholder=\\"Enter color\\" id=\\"formGridPrice\\" class=\\"bg-dark text-white form-control\\"/></div></div><div class=\\"form-row\\"><div class=\\"form-group col\\"><label class=\\"form-label\\" for=\\"formGridRam_rom\\"> Phone&#x27;s price </label><input type=\\"text\\" required=\\"\\" autoComplete=\\"off\\" name=\\"price\\" value=\\"\\" placeholder=\\"Enter your phone&#x27;s price\\" id=\\"formGridRam_rom\\" class=\\"bg-dark text-white form-control\\"/></div><div class=\\"form-group col\\"><label class=\\"form-label\\" for=\\"formGridImageL\\"> Phone&#x27;s version </label><input type=\\"text\\" required=\\"\\" autoComplete=\\"off\\" name=\\"version\\" value=\\"\\" placeholder=\\"Enter version \\" id=\\"formGridImageL\\" class=\\"bg-dark text-white form-control\\"/></div></div><div class=\\"form-row\\"><div class=\\"form-group col\\"><label class=\\"form-label\\" for=\\"formGridDescription\\"> Choose in which product your phone fits </label><br/><label>Choose ProductName:</label><select id=\\"select\\"></select></div></div></div><div class=\\"card-footer\\" style=\\"text-align:center\\"><button type=\\"submit\\" class=\\"btn btn-success btn-sm\\">Sell Phone</button>  <button type=\\"reset\\" class=\\"btn btn-info btn-sm\\">Reset</button></div></form>"`;
